jsonnet/thanos-mixin/dashboards/receive.libsonnet

```diff
@@ -0,0 +1,257 @@
+            'all',
+          )
+        )
+        .addPanel(
+          g.panel('Errors') +
+          g.qpsErrTotalPanel(
+            'thanos_receive_forward_requests_total{namespace="$namespace",%(thanosReceiveSelector)s,result="error"}' % $._config,
+            'thanos_receive_forward_requests_total{namespace="$namespace",%(thanosReceiveSelector)s}' % $._config,
+          )
+        )
+      )
+      .addRow(
+        g.row('gRPC (Unary)')
+        .addPanel(
+          g.panel('Rate') +
+          g.grpcQpsPanel('server', 'namespace="$namespace",%(thanosReceiveSelector)s,grpc_type="unary"' % $._config)
+        )
+        .addPanel(
+          g.panel('Errors') +
+          g.grpcErrorsPanel('server', 'namespace="$namespace",%(thanosReceiveSelector)s,grpc_type="unary"' % $._config)
+        )
+        .addPanel(
+          g.panel('Duration') +
+          g.grpcLatencyPanel('server', 'namespace="$namespace",%(thanosReceiveSelector)s,grpc_type="unary"' % $._config)
+        )
+      )
+      .addRow(
+        g.row('Detailed')
+        .addPanel(
+          g.panel('Rate') +
+          g.grpcQpsPanelDetailed('server', 'namespace="$namespace",%(thanosReceiveSelector)s,grpc_type="unary"' % $._config)
+        )
+        .addPanel(
+          g.panel('Errors') +
+          g.grpcErrDetailsPanel('server', 'namespace="$namespace",%(thanosReceiveSelector)s,grpc_type="unary"' % $._config)
+        )
+        .addPanel(
+          g.panel('Duration') +
+          g.grpcLatencyPanelDetailed('server', 'namespace="$namespace",%(thanosReceiveSelector)s,grpc_type="unary"' % $._config)
+        ) +
+        g.collapse
+      )
+      .addRow(
+        g.row('gRPC (Stream)')
+        .addPanel(
+          g.panel('Rate') +
+          g.grpcQpsPanel('server', 'namespace="$namespace",%(thanosReceiveSelector)s,grpc_type="server_stream"' % $._config)
+        )
+        .addPanel(
+          g.panel('Errors') +
+          g.grpcErrorsPanel('server', 'namespace="$namespace",%(thanosReceiveSelector)s,grpc_type="server_stream"' % $._config)
+        )
+        .addPanel(
+          g.panel('Duration') +
+          g.grpcLatencyPanel('server', 'namespace="$namespace",%(thanosReceiveSelector)s,grpc_type="server_stream"' % $._config)
+        )
+      )
+      .addRow(
+        g.row('Detailed')
+        .addPanel(
+          g.panel('Rate') +
+          g.grpcQpsPanelDetailed('server', 'namespace="$namespace",%(thanosReceiveSelector)s,grpc_type="server_stream"' % $._config)
+        )
+        .addPanel(
+          g.panel('Errors') +
+          g.grpcErrDetailsPanel('server', 'namespace="$namespace",%(thanosReceiveSelector)s,grpc_type="server_stream"' % $._config)
+        )
+        .addPanel(
+          g.panel('Duration') +
+          g.grpcLatencyPanelDetailed('server', 'namespace="$namespace",%(thanosReceiveSelector)s,grpc_type="server_stream"' % $._config)
+        ) +
+        g.collapse
+      )
+      .addRow(
+        g.row('Hashring Status')
```

Comment:
I think it would make sense to separate these hashring dashboards from the Thanos mixin. This mixin is only concerned with the Thanos receive component, whereas the hashrings are composed of thanos receive and the thanos-receive-controller creating a more high level domain object.

Reply:
But I like them a lot and would love to add a them as a separate dashboard in https://github.com/observatorium/configuration/ :+1:

Reply:
I'll remove them from `thanos-mixin` and create a separate PR for configuration.